### ISSUE TITLE
Google Speech-To-Text API

### ISSUE THREAD

**Issue by klevismino** [ratwithacompiler/OBS-captions-plugin#57]
Hello 

Congratulations on the great work you have done!
I also checked your [reddit post](https://www.reddit.com/r/Twitch/comments/chrgqc/i_made_a_obs_plugin_for_closed_captioning_using/euzyox8/?utm_source=share&utm_medium=web2x&context=3).
Do you plan to support Google's official API for [Speech-To-Text](https://cloud.google.com/speech-to-text/docs/how-to)?
You could add that as an option, where the user can provide his GCP API key.

Thanks and best regards
Klevis

**Comment by ratwithacompiler** [comment]
Hi

Yeah I already implemented support for the GRPC API a while ago, there's a working (but now outdated) version in the #grpc-public branch and I've kept it up to date locally.
I was planning to transition to that fully a while ago but things got held up, but it should happen eventually (hopefully).

**Comment by WiredUK** [comment]
Is there any way we can help get this moved along a bit? I think the current API is rate-limiting or something as it keeps stopping. Is there a way we can contribute somehow?

PS Thanks for the excellent work though, it's superb.

**Comment by WiredUK** [comment]
I have tried to build your GRPC branch but it relies on private projects in DevOps so I'm stuck. I wanted to give it a go and embed my own API key into it but I'm at a brick wall. I'm not kidding that we (https://sitp.online) would contribute some money to you if we could get this plugin to work reliably. We're a volunteer organisation but could spare a small amount of money to help.

**Comment by ratwithacompiler** [comment]
Yeah I've been working on that over the last couple days, updated version is now on the main branch here. Setting up working Windows builds involving GRPC is always quite the hassle unfortunately.

Windows builds are still not fully working yet and the current code also requires OBS functions for virtualcam support that aren't in any stable release yet so currently it only works with an OBS compiled with those changes until OBS 27 comes out.

But if you do really wanna run it you can get OBS builds from current master branch from their CI builds.

Latest one: https://github.com/obsproject/obs-studio/runs/2063443158 and just download the winbuild there 
https://dev.azure.com/obsjim/obsjim/_build/results?buildId=6251&view=artifacts&pathAsName=false&type=publishedArtifacts

Static build of the plugin with the GRPC API: [obs_google_caption_plugin windows grpc static testing.zip](https://github.com/ratwithacompiler/OBS-captions-plugin/files/6112289/obs_google_caption_plugin.windows.grpc.static.testing.zip)
which should in theory work with that (did on my machine at least but who knows). Just has a API Key value in the settings.

**Comment by ratwithacompiler** [comment]
Ohh actually nevermind about needing a brand new OBS version, this one should work in official OBS 26.1.1: 
[obs_google_caption_plugin groc static 26.1.1.zip](https://github.com/ratwithacompiler/OBS-captions-plugin/files/6112538/obs_google_caption_plugin.groc.static.26.1.1.zip)

Just commented out the virtucalcam stuff relying on new OBS real quick but should otherwise work fine.

**Comment by WiredUK** [comment]
Works like a charm so far, will give it some more testing when I get a chance (maybe tonight) and report back. Thank you for being so responsive, it's really appreciated! Do you have a ko-fi or something like that set up? I'd like to, at the very least, buy you a coffee!

**Comment by WiredUK** [comment]
Oh, and if you have the Linux version too, that would be fantastic!

**Comment by ratwithacompiler** [comment]
No problem! Yeah I kinda have a ko-fi https://ko-fi.com/ratuin but it's paypal only since I never got around to setting up stripe.

For Linux and Mac the CI builds for the GRPC versions should just work in theory: https://dev.azure.com/ratwithacompiler/OBS-captions-plugin/_build/results?buildId=677&view=artifacts&pathAsName=false&type=publishedArtifacts ,only Windows is a bit more annoying with loading in older versions of OBS.

**Comment by WiredUK** [comment]
We could do a direct PayPal if you prefer that? Going to get you some money tonight, you've earned it!

**Comment by ratwithacompiler** [comment]
Thanks, not necessary but very much appreciated! I think Ko-Fi goes directly from paypal to paypal without them in the middle already to get the 0% fee thing the advertise so should be the same I believe.

**Comment by WiredUK** [comment]
OK, you should have a bit of money in there now from SitPO. Enjoy!

**Comment by ratwithacompiler** [comment]
@WiredUK Ohh wow. Got it. Thank you very much!! Helps a lot!

**Comment by esteinborn** [comment]
Hi, Upgraded to OBS 27 this morning saw that there was also an update to this plugin. After updating it, my API field was removed. I then remembered this thread, and added the DLL back in. Everything works fine, but I wanted to query about whether we can get this as a packaged version for v27 with the API field intact?

Great plugin btw!

**Comment by ratwithacompiler** [comment]
You can get the GRPC builds with key input for all versions from the CI but it's a bit hard to navigate on Github, findable via the commits section and the green CI checkmarks and following the links to Azure CI.

But probably easiest to get them from here: https://dev.azure.com/ratwithacompiler/OBS-captions-plugin/_build?definitionId=7
Click the latest version and then `Artifacts` and then download the GRPC version for you platform.

Direct link for 27, you'll want`[platform]_GRPC` not the http ones,  just download the .zip
https://dev.azure.com/ratwithacompiler/OBS-captions-plugin/_build/results?buildId=712&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by BtbN** [comment]
How do you deal with the cost for this? Doing some calculations, I end up that using this plugin 24/7 for a month would rack up south of 1000$ of charges. For a single stream only, assuming it's 24/7 and using the enhanced video model.

**Comment by APaulSys** [comment]
Are these GRPC builds still being offered?  I'm not seeing artifacts for the latest versions in the links above.

**Comment by ratwithacompiler** [comment]
@APaulSys yeah GRPC versions are part of all builds still but looks like the artifacts expired after 3 months.

Ran new ones, now https://dev.azure.com/ratwithacompiler/OBS-captions-plugin/_build/results?buildId=858&view=artifacts&type=publishedArtifacts

**Comment by theroboben** [comment]
Any possibility of getting another azure run?
